carma-messenger-core/cpp_message/src/SDSM_Message.cpp

```diff
@@ -1072,4 +1072,958 @@ namespace cpp_message
 
     }
 
+
+
+    // SDSM decoding
+    boost::optional<j3224_v2x_msgs::msg::SensorDataSharingMessage> SDSM_Message::decode_sdsm_message(std::vector<uint8_t>&binary_array){
+        
+        j3224_v2x_msgs::msg::SensorDataSharingMessage output;
+        // decode results - stored in binary_array
+        asn_dec_rval_t rval;
+        MessageFrame_t* message = nullptr;
+        
+        // copy from vector to array         
+        size_t len = binary_array.size();    
+        uint8_t buf[len];             
+        std::copy(binary_array.begin(),binary_array.end(),buf);
```

Comment:
Why are we using a raw array here? We should be using `std::array` or `std::vector`.

Reply:
uper uses raw array input from what I can tell